### DISCUSSION TITLE
Difference to grinder

### DISCUSSION THREAD

**Discussion by enyo** [frencojobs/derry#72] [Q&A] [answered]
It would be great to get some information on how this package differs from `grinder` and what the advantage of `derry` is.
I think it would make sense to add a short note about that in the README since it covers similar functionality.

**Comment by frencojobs** [comment, marked as answer]
Grinder is more configurable as it has a dedicated config system that has to be written in Dart, bringing the advantages of being able to use Dart's libraries directly in the config. But it is definitely more work to do when you want is just to make a shortcut for a specific script. 

If you're coming from a NodeJS background, derry behaves pretty much the same as NPM scripts except for the `pre` and `post` scripts, (which I will add to derry very soon).

**Edit**: support for `pre` & `post` scripts has been added as of _29-12-2020_.

Derry helps you to write less redundant scripts. As written in [this](https://www.freecodecamp.org/news/introduction-to-npm-scripts-1dbb2ae01633/) article about NPM scripts, it lets you avoid the need for tooling and configurations. It helps you use well-managed scripts, and it is something I wish `pub` would support natively. With the hope that if `pub` would support scripts natively someday,  I even named the derry config object `scripts` to make it easier to migrate.
```yaml
scripts: 
# ^ this should be named "derry" actually, which is common and as guided
# but I named it "scripts" in case pub would support scripts someday
# like `pub run build`
  build: pub run build_runner build --delete-conflicting-outputs
```
If what you need to do is to write a config for a very extensive task and to share it between CI/automation, and your development, [grinder](https://pub.dev/packages/grinder) is very configurable and could be very useful. But if what you need is just to define shortcut scripts without the hassle of configurations, Derry could be a hand. As written in this [article](https://css-tricks.com/why-npm-scripts/) even though NPM scripts are very handy and cover most of the use cases most of the time during development, it's not like you're obligated to use them. 
> if you are happy with your current build system and it accomplishes all that you need it to do, you can keep using it!
>
> from https://css-tricks.com/why-npm-scripts

I believe both grinder and derry documentations are well written, a user could probably read both and decide what's the best for him/her.

> a short note about that in the README

About that, I'll probably leave this issue open for people to see. Thank you for taking the time to open this issue, I appreciate it ☺️!

**Comment by enyo** [comment]
Perfect answer! Thanks.

**Comment by fredgrott** [comment]
Okay, grinder allows one to run other dart binary tools that you may have added to automate some item of your project workflow.  In fact, one can use Derry, build_runnner, and grinder together. I am switching to such a toolchain. Keep in mind that we no longer have Pub.build task however when using Grinder
